src/service/viewport-impl.js

```diff
@@ -290,10 +296,11 @@ export class Viewport {
 
   /**
    * Returns the size of the viewport.
+   * @param {SizeReportingFidelity} fidelity
    * @return {!{width: number, height: number}}
    */
-  getSize() {
-    if (this.size_) {
+  getSize(fidelity = SizeReportingFidelity.LOW) {
+    if (this.size_ && fidelity == SizeReportingFidelity.LOW) {
```

Comment:
hmm `vp.(onResize|onChange)( () => vp.getSize() )` should always be correct, if it is not, let's treat it as a bug and fix it.